pkg/cloud/provider/provider.go

```diff
@@ -149,12 +149,20 @@ func ShareTenancyCosts(p models.Provider) bool {
 // NewProvider looks at the nodespec or provider metadata server to decide which provider to instantiate.
 func NewProvider(cache clustercache.ClusterCache, apiKey string, config *config.ConfigFileManager) (models.Provider, error) {
 	nodes := cache.GetAllNodes()
-	if len(nodes) == 0 {
-		log.Infof("Could not locate any nodes for cluster.") // valid in ETL readonly mode
-		return &CustomProvider{
-			Clientset: cache,
-			Config:    NewProviderConfig(config, "default.json"),
-		}, nil
+	var numRetries int
```

Comment:
This is kinda gross, and I'd love some thorough feedback on it. It worked for the customer in a pinch, but it doesn't seem _good_. Would love for there to be a guarantee that `cache.GetAllNodes()` will block until it has retrieved all available nodes. What's super interesting is that we only ever actually use the first node in the nodes list 🤔

Reply:
Do we still need this code?

Reply:
unfortunately, we did confirm that we needed this code, even with your PR from yesterday merged

Reply:
we had first tried with just your PR contents, but we still had the same error where nothing was coming back and we were defaulting to the default provider

Reply:
There is some deeper bug here, because I did confirm the initial issue. Let me find where this provider is initialized.